### PR TITLE
ARROW-10333: [Java] Get rid of org.apache.arrow.util in vector

### PR DESCRIPTION
**Pull request by HedgehogCode**
The package `org.apache.arrow.util` was present in the artifacts arrow-vector and arrow-vector-memory-core. Split packages are a problem for OSGI and the Java 9+ modules (JPMS).

This PR moves the classes `AutoClosables` and `Collections2` to arrow-memory-core because they are generally useful and have no coupling with arrow-vector.
The class `DataSizeRoundingUtil` is tighter coupled with arrow-vector and is therefore moved to the package `org.apache.arrow.vector.util`.

### CODE REVIEW

File: java/vector/src/main/codegen/templates/DenseUnionVector.java [hunk 1]
```diff
@@ -36,6 +35,7 @@
 import org.apache.arrow.vector.types.pojo.ArrowType;
 import org.apache.arrow.vector.types.pojo.Field;
 import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.util.DataSizeRoundingUtil;
```
Comment (liyafan82) [inline java/vector/src/main/codegen/templates/DenseUnionVector.java:38]: This line should be placed after the next line, to maintain the import order

Comment (HedgehogCode) [reply java/vector/src/main/codegen/templates/DenseUnionVector.java:38]: True. I changed it.

Comment (liyafan82) [reply java/vector/src/main/codegen/templates/DenseUnionVector.java:38]: We need to import it only once

Comment (HedgehogCode) [reply java/vector/src/main/codegen/templates/DenseUnionVector.java]: Sorry about that. Must have happened during the rebasing of my local changes. It's finally fixed now.

**Review by liyafan82 — APPROVED** [review]
LGTM, will merge soon if there are no more comments.

**Comment by liyafan82** [comment]
Merging. Thanks for the patch @HedgehogCode

**Pull request closed without merging**